node/as/README.md

```diff
@@ -0,0 +1,215 @@
+# TChannel as handlers.
+
+The `as` handlers are a high level interface for sending
+and receiving messages over tchannel with non-buffer encoding.
+
+There are currently two `as` handlers, `json` and `thrift`. 
+Each one sets the `as` transport header to their respective type.
+
+## as send interface
+
+To make requests to a `json` or `thrift` service using the
+tchannel request interface looks like.
+
+```js
+tchannel
+    .request({
+        serviceName: 'foo',
+        timeout: 1000
+    })
+    .send(
+        Encoder.stringify(arg1),
+        Encoder.stringify(arg2),
+        Encoder.stringify(arg3),
+        function onResponse(err, resp) {
+            if (err) {
+                console.log('got error', {
+                    error: err
+                })
+                return;
+            }
+
+            if (!resp.ok) {
+                console.log('resp not ok', {
+                    arg2: Encoder.parse(resp.arg2),
+                    arg3: Encoder.parse(resp.arg3)
+                });
+                return;
+            }
+
+            console.log('resp ok', {
+                arg2: Encoder.parse(resp.arg2),
+                arg3: Encoder.parse(resp.arg3)
+            });
+        }
+    );
+```
+
+The `as` client interface can be made simpler then this.
+
+The proposed interface is modeled after `request/request` which
+is a popular HTTP client with a simpler interface.
+
+```js
+var TChannelJSON = require('tchannel/as/json');
+var tchannelJSON = TChannelJSON({
```

Comment:
can we please not have to instantiate a thing? all this injected stuff can be had off the channel.... if req.channel isn't yet a thing, then it can be made so